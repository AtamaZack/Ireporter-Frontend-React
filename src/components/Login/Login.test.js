import React from 'react';
import { shallow } from 'enzyme';
import { Login , mapStateToProps} from './index';

describe('<Login />', () => {
  let wrapper;
  const props = {
    loginUser: jest.fn()
  };
  beforeEach(()=>{
    wrapper = shallow(<Login {...props}/>);
  })

  it('should render the signup component', () => {  
    expect(wrapper).toMatchSnapshot();
  }); 

  it('handle change', ()=>{
    const e = {
      target:{
        name:"email",
        value:"zack@test.com"
      }
    }
    wrapper.instance().handleChange(e);
    expect(wrapper.instance().state.email).toBe('zack@test.com');
  });

  it('handle submit', ()=>{
    wrapper.instance().handleSubmit({ preventDefault:jest.fn() });
    expect(props.loginUser).toHaveBeenCalled();
  });

  it('mapStateToProps', ()=>{
    const initialState = { loginReducer:{user: {},}
    };
    expect(mapStateToProps(initialState)).toEqual({user: {}})

  });
});

