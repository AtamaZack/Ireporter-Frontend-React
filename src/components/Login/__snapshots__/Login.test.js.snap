// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render the signup component 1`] = `ShallowWrapper {}`;
